pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -146,6 +147,13 @@ func (c *Reconciler) reconcilePA(ctx context.Context, rev *v1alpha1.Revision) er
 			logger.Errorf("Error creating PA %s: %v", paName, err)
 			return err
 		}
+		/* TODO #5403 once after confirmation of number of classes autoscaler can support than this condition can be removed and handled through validation.
+		Currently when given class value is other than (KPA and HPA) /metric value will be empty
+		so based on the empty value of metric we can decide whether provided class is valid or invalid*/
+		if value, ok := pa.Annotations[autoscaling.MetricAnnotationKey]; ok && value == "" {
```

Comment:
I don't think this is accurate. If this value is empty there might still be a third-party controller slated to pick this up. All we can do for validation here really is to check if there's no other value in the autoscaling.serving... namespace. Other than that, it's basically works as designed i think.

Reply:
Also, validation should be handled by [annotation_validation.go](https://github.com/knative/serving/blob/master/pkg/apis/autoscaling/annotation_validation.go#L119)

Reply:
@taragu We don't have concrete values for `autoscaling.knative.dev/class` annotation currently only KPA and HPA supported but it can be extended so that's the reason dint handle in validation part for more info #5403

Reply:
@markusthoemmes can you please elaborate little bit more on this.
like
` If this value is empty there might still be a third-party controller slated to pick this up`
`if there's no other value in the autoscaling.serving... namespace.`

Reply:
Well as was discussed in the issue I think, there is no way for us to tell statically if a the /class/metric tuple is valid or not. An empty metric might just be defaulted by another autoscaler to something sensible, so that's no indication either.

What we can do though, is validate that, when using knative's annotation namespace, only hpa or kpa are valid.

Reply:
I agree with Markus and Tara's points.  I'm not sure it is possible to do this statically, but the webhook is a better place for it than this.

Reply:
@mattmoor Yes I agree.
@markusthoemmes and me discussed and decided to go with this approach

**Approach:**

1. Implement a status called `Reconciled` and each PA should immediately set to true.
2. Then based on the `Reconciled` status we can indicate the provided /class is supported by knative or not
3. Then we can show proper error message saying why revision is failed, Because currently we don't get proper reason why revision failed.

So this way we can avoid validation of /class annotation because as discussed in the issue we don't have concrete values for the class.
And in future if any new PA added that should implement `Reconciled`

Reply:
A bit more precisely: We can use that condition to indicate that nothing has reconciled the PA yet. That can then indicate to the user, that something might be wrong on that end.

We cannot really say it's not supported as you've shown in your example. I'd alter it to say something like `PodAutoscaler with class "foo.bar" has not been reconciled yet" to indicate exactly what we know.